src/charm.py

```diff
@@ -6,36 +6,43 @@
 """A machine charm for Vault."""
 
 import logging
-from typing import Dict, List, Optional
+from contextlib import contextmanager
+from typing import Dict, List, Optional, Tuple
 
 import hcl
 from charms.grafana_agent.v0.cos_agent import COSAgentProvider
 from charms.operator_libs_linux.v2 import snap
+from charms.vault_k8s.v0.vault_client import AppRole, AuditDeviceType, Token, Vault
 from charms.vault_k8s.v0.vault_tls import (
     File,
     VaultTLSManager,
 )
 from jinja2 import Environment, FileSystemLoader
 from machine import Machine
+from ops import ActionEvent, BlockedStatus, SecretNotFoundError
 from ops.charm import CharmBase, CollectStatusEvent
 from ops.main import main
 from ops.model import ActiveStatus, MaintenanceStatus, ModelError, WaitingStatus
 
 logger = logging.getLogger(__name__)
 
+CHARM_POLICY_NAME = "charm-access"
+CHARM_POLICY_PATH = "src/templates/charm_policy.hcl"
 CONFIG_TEMPLATE_DIR_PATH = "src/templates/"
 CONFIG_TEMPLATE_NAME = "vault.hcl.j2"
 PEER_RELATION_NAME = "vault-peers"
+VAULT_CHARM_APPROLE_SECRET_LABEL = "vault-approle-auth-details"
 VAULT_CONFIG_PATH = "/var/snap/vault/common"
 VAULT_CONFIG_FILE_NAME = "vault.hcl"
+VAULT_DEFAULT_POLICY_NAME = "default"
 VAULT_PORT = 8200
 VAULT_CLUSTER_PORT = 8201
 VAULT_SNAP_NAME = "vault"
-VAULT_SERVICE_NAME = "vault"
+VAULT_SERVICE_NAME = "vaultd-start"
```

Comment:
It seems that @Kayra1 has run into this issue now. It may be a little more nuanced than I described above.

Since we're now also experiencing this issue in the k8s version of the charm, it makes sense to move ahead with the `vault_tls` changes.

This should go away once those changes are made.

Reply:
ref https://github.com/canonical/vault-k8s-operator/pull/253